spec/ruby_snowflake/client_spec.rb

```diff
@@ -23,14 +23,58 @@
   end
 
   describe "querying" do
-    let(:query) { "" }
-    let(:result) { client.query(query) }
+    subject(:result) { client.query(query, query_name: "test_query") }
+    let(:query) { "SELECT 1;" }
 
-    context "with the alias name `fetch`" do
-      let(:query) { "SELECT 1;" }
-      it "should work" do
-        result = client.fetch(query)
+    it "emits instrumentation events" do
+      start_event_received = false
+      finish_event_received = false
+      subscriptions = []
+      subscriptions << ActiveSupport::Notifications.subscribe("rb_snowflake_client.snowflake_query.start") do |event|
+        expect(event.payload[:query_name]).to eq("test_query")
+        start_event_received = true
+      end
+      subscriptions << ActiveSupport::Notifications.subscribe("rb_snowflake_client.snowflake_query.finish") do |event|
+        expect(event.payload[:query_name]).to eq("test_query")
+        finish_event_received = true
+      end
+
+      begin
+        expect(result).to be_a(RubySnowflake::Result)
+      ensure
+        subscriptions.each { ActiveSupport::Notifications.unsubscribe(_1) }
+      end
+
+      expect(start_event_received).to be(true)
+      expect(finish_event_received).to be(true)
+    end
+
+    context "with the 'fetch' alias" do
+      subject(:result) { client.fetch(query) }
+
+      it "works with 'fetch' alias" do
+        expect(result).to be_a(RubySnowflake::Result)
+        expect(result.length).to eq(1)
+        rows = result.get_all_rows
+        expect(rows).to eq(
+          [{"1" => 1}]
+        )
+      end
+    end
+
+    context "without ActiveSupport" do
+      around do |example|
+        active_support = ::ActiveSupport
+        ActiveSupport = nil
+
+        begin
+          example.run
+        ensure
+          ActiveSupport = active_support
+        end
+      end
```

Comment:
We can use `stub_const` here:

```suggestion
    around do |example|
      if defined?(ActiveSupport)
        stub_const("ActiveSupport", nil)
      end
      example.run
    end
```

Reply:
fixed!